lib/internal/test_runner/test.js

```diff
@@ -204,6 +210,12 @@ class Test extends AsyncResource {
       this.testNumber = parent.subtests.length + 1;
       this.timeout = parent.timeout;
       this.root = parent.root;
+      this.hooks = {
+        before: [],
+        after: [],
```

Comment:
How are these copied to deeply nested test suites?

Reply:
@rochdev the purpose of comments on a PR are for reviewing changes, general questions about the code can be pretty distracting

Reply:
The question was about a potential bug in the code. If these are not copied to nested test suites, then this code doesn't work.